### PR TITLE
Fix docker deploy tagging

### PR DESCRIPTION
**Pull request by kaitj**
## Proposed changes

The tags that were being used was determined based on the actions `event` in the metadata step. 
(for reference: https://github.com/marketplace/actions/docker-metadata-action)

This PR moves the deploy action to its own workflow, acting on pushing of tags in order to push only the tagged version and `latest` to docker hub. Additionally, this PR "fixes" how the docker container was built, creating a wheel and installing it in a separate step to expose the labelmerge CLI.

## Types of changes

What types of changes does your code introduce? Put an `x` in the boxes that apply

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionalitiy)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [ ] Changes have been tested to ensure that fix is effective or that a feature works.
- [ ] Changes pass the unit tests
- [ ] Code has been run through the `poe quality` task
- [ ] I have included necessary documentation or comments (as necessary)
- [ ] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,11 +11,14 @@ RUN mkdir -p /opt \
...
+    && poetry build -f wheel \
     && apt-get purge -y -q g++ \
     && apt-get --purge -y -qq autoremove 
     
 # Stage: runtime
 FROM build AS runtime
-COPY ./labelmerge /opt/labelmerge
-ENTRYPOINT ["/opt/labelmerge/run.py"]
\ No newline at end of file
+RUN WHEEL=`ls /opt/labelmerge/dist/* | grep whl` \
```
Comment (tkkuehn) [inline Dockerfile:21]: With the caveat that getting the path to the wheel might be trickier than this, the image would be a bit smaller if we did something like:

```
FROM python:3.9-slim-bullseye AS runtime
COPY --from=build /opt/labelmerge/dist/*.whl /opt/labelmerge
RUN WHEEL=`ls /opt/labelmerge | grep whl` \
    && pip install $WHEEL \
    && rm -r /opt/labelmerge
ENTRYPOINT ["labelmerge"]
```

File: pyproject.toml [hunk 2]
```diff
@@ -12,6 +12,9 @@ authors = [
     "Ali R. Khan"
 ]
 readme = "README.md"
+packages = [
```
Comment (tkkuehn) [inline pyproject.toml:15]: Is this necessary? I would have expected poetry to be able to automatically pick it up.

**Review by tkkuehn — CHANGES_REQUESTED** [review]
A couple of minor changes that you don't necessarily need to make

File: .github/workflows/deploy_docker.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+---
+name: Deploy docker container
+
+on:
+  release:
```
[the remark this answers is not in the document]
Comment (kaitj) [reply .github/workflows/deploy_docker.yml:5]: Yeah, that would work too - I had something set up similar to what you had originally suggested, but with tags: `- v*`. I ended up opting to use the release trigger instead as it was used in a few other workflows and effectively does the same thing.

File: pyproject.toml (hunk 2 above)
Comment (kaitj) [reply pyproject.toml:15]: I tried to do it without this, but it didn't seem to pick it up in my tests.

File: Dockerfile (hunk 1 above)
Comment (kaitj) [reply Dockerfile:21]: Can definitely give this a try on Monday, this container is easy and quick enough to build.

File: pyproject.toml (hunk 2 above)
Comment (tkkuehn) [reply pyproject.toml:15]: That's odd but if this works I'm fine with it

File: .github/workflows/deploy_docker.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tkkuehn) [reply .github/workflows/deploy_docker.yml:5]: Ah it's not really documented well that it'll do the same thing but seems like you're right!

**Review by tkkuehn — APPROVED** [review]

**Pull request merged**